src/components/reusables/LogInForm.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from "react";
+import { connect } from "react-redux";
+import { Store } from "redux";
+import { AdminData } from "../../interfaces";
+import Form from "./Form";
+import Fieldset from "./Fieldset";
+import Input from "./Input";
+import ActionCreator from "../../actions";
+import { State } from "../../reducers/index";
+
+export interface LogInFormStateProps {
+  admin?: AdminData;
+}
+
+export interface LogInFormOwnProps {
+  extraFields?: Array<JSX.Element>;
+  title?: string;
+  legend?: string;
+}
+
+export interface LogInFormDispatchProps {
+  logIn: (data: FormData) => Promise<void>;
+}
+
+export interface LogInFormContext {
+  store: Store<State>;
+}
+
+export interface LogInFormProps extends LogInFormStateProps, LogInFormOwnProps, LogInFormDispatchProps {};
+
+export class LogInForm extends React.Component<LogInFormProps, void> {
+
+  context: LogInFormContext;
+  static contextTypes: React.ValidationMap<LogInFormContext> = {
+    store: React.PropTypes.object.isRequired
+  };
+
+  constructor(props: LogInFormProps) {
+    super(props);
+    this.submit = this.submit.bind(this);
+  }
+
+  async submit(data: FormData): Promise<void> {
+    await this.props.logIn(data);
+    window.location.reload();
+  }
+
+  render(): JSX.Element {
+    let title = this.props.title ? this.props.title : "Log In";
+    let legend = this.props.legend ? this.props.legend : "Credentials";
+
+    let username = <Input key="username" name="username" label="Username" />;
+    let password = <Input key="password" type="password" name="password" label="Password" />;
+    let elements = this.props.extraFields ? [username, password].concat(this.props.extraFields) : [username, password];
+    let fieldset = <Fieldset key="credentials" legend={legend} elements={elements} />;
```

Comment:
Why do you need these `key` props here?

Reply:
Because React becomes distressed if child components don't have unique `key` props.

Reply:
Hmm I understand for the username and password for the array but not for the fieldset. I don't get any errors when removing the `key` prop.

Reply:
I get errors when I remove it.  (Try logging out.)

Reply:
Hmm yeah I see them.